plugin/src/main/groovy/com/akaita/android/easylauncher/filter/ColorRibbonFilter.java

```diff
@@ -86,15 +166,24 @@ public void apply(BufferedImage image) {
 
             FontMetrics fm = g.getFontMetrics();
 
-            drawString(g, label,
-                    (int) -labelBounds.getWidth() / 2,
-                    y + fm.getAscent());
+            if (position == LayoutPosition.TopHorizontal || position == LayoutPosition.BottomHorizontal) {
+                drawString(g, label,
+                        (width / 2) - ((int) labelBounds.getWidth() / 2),
+                        y + fm.getAscent() + padding);
+            } else {
+                drawString(g, label,
+                        (int) -labelBounds.getWidth() / 2,
+                        y + fm.getAscent());
+            }
+
         }
         g.dispose();
     }
 
-    private Font getFont(int maxLabelWidth, FontRenderContext frc) {
-        int max = largeRibbon ? 64 : 32;
+    private Font getFont(int iconHeight, int maxLabelWidth, FontRenderContext frc) {
+//        int max = largeRibbon ? 64 : 32;
+//        int max = largeRibbon ? 45 : 32;
+        int max = iconHeight / 8;
```

Comment:
This is actually something I mentioned in https://github.com/akaita/easylauncher-gradle-plugin/pull/24 . I think this is the right path to take